sdk/common/src/main/java/io/opentelemetry/sdk/resources/Resource.java

```diff
@@ -36,11 +36,15 @@
           + " characters.";
   private static final String ERROR_MESSAGE_INVALID_VALUE =
       " should be a ASCII string with a length not exceed " + MAX_LENGTH + " characters.";
-  private static final Resource EMPTY = create(Attributes.empty());
 
   private static final Resource TELEMETRY_SDK;
+  private static final String FALLBACK_SERVICE_NAME;
 
   static {
+    String osName = System.getProperty("os.name");
+    boolean windows = osName != null && osName.toLowerCase().contains("windows");
+    FALLBACK_SERVICE_NAME = windows ? "unknown_service:java.exe" : "unknown_service:java";
```

Comment:
Can we leave off `java`? The spec says if we don't have access to the executable name, we don't need to include it. Since we're basically taking a random guess here, I guess it means we don't have access to the executable name?

Reply:
well, it's not *quite* a random guess. We know that the code is at least running on a JVM (graal native binaries aside for now), "java" is at least more descriptive than nothing. I'd be amenable to just hard-coding "unknown_service:java" so it  is at least known that the thing they've misconfigured is a java service. It would narrow down the search and at least eliminate non-java apps when trying to find which app was misconfigured.

Reply:
(it would also make the testing easier to just have something hard-coded, so I'm not opposed to leaving windows users without their `.exe`)

Reply:
The constant seems fine to me too, removing doesn't seem that bad function-wise since we have the language in the resource already, but either works

Reply:
I think the main class name would make much more sense for java than the exe name. How precise/strict is the spec here? Would that be allowed?

Reply:
The spec says executable so I read it as currently no, but it would be simple to add a clarification. It would probably bring all script languages into play, e.g., python main.py should have main.py but what about /foo/bar/main.py. Would we want the fully qualified name here? It seems possibly too long, though simple name seems too short.

Pretty complicated, I wish not setting service.name could be a startup exception 😨